insights/client/config.py

```diff
@@ -487,10 +487,20 @@ def load_config_file(self, fname=None):
             pass
         d = dict(parsedconfig.items(constants.app_name))
         for key in d:
-            if key == 'retries':
-                d[key] = parsedconfig.getint(constants.app_name, key)
-            if key in DEFAULT_BOOLS and isinstance(d[key], six.string_types):
-                d[key] = parsedconfig.getboolean(constants.app_name, key)
+            try:
+                if key == 'retries':
+                    d[key] = parsedconfig.getint(constants.app_name, key)
+                if key == 'http_timeout':
+                    d[key] = parsedconfig.getfloat(constants.app_name, key)
+                if key in DEFAULT_BOOLS and isinstance(
+                        d[key], six.string_types):
+                    d[key] = parsedconfig.getboolean(constants.app_name, key)
+            except ValueError as e:
+                if self._print_errors:
+                    sys.stdout.write(
+                        'ERROR: {0}.\nCould not read configuration file, '
+                        'using defaults\n'.format(e))
+                return
```

Comment:
So returning here means it just stops processing the remaining keys?  Seems like it should either halt the whole process or keep going.

Reply:
The config object isn't updated with the values from the .conf file until the very end of the function. So if any exception forces a return it won't be changed from the defaults. We've never supported a case where we only parse _some_ of the config vars from the file.

Reply:
Makes sense, thanks.